c10/xpu/XPUFunctions.cpp

```diff
@@ -39,15 +40,20 @@ void enumDevices(std::vector<std::unique_ptr<sycl::device>>& devices) {
     auto device_list = platform.get_devices();
     for (const auto& device : device_list) {
       if (device.is_gpu()) {
-        devices.push_back(std::make_unique<sycl::device>(device));
+        gDevicePool.devices.push_back(std::make_unique<sycl::device>(device));
+        if (device.has(sycl::aspect::ext_intel_device_info_uuid))
+          gDevicePool.uuids.push_back(
+              device.get_info<sycl::ext::intel::info::device::uuid>());
+        else
+          gDevicePool.uuids.push_back(std::array<unsigned char, 16>{});
```

Comment:
in which case there will be no uuid?
if no uuid, we should raise a warning, or not?
what's the impact?

Reply:
A warning should be thrown when no uuid. PTI provides all profiling data including the device uuid and xpu kineto needs to save the running device into the kernel activity. The device is fetched from the global device pool by the uuid.

Reply:
Let's give a warning to user if no UUID

Reply:
> Let's give a warning to user if no UUID

ok